### PR TITLE
Return only channels for query string starting with '@'

### PR DESCRIPTION
**Pull request by ioancole**
For query string starting with '@', return only list of matching channels.
Else return list of results as before.

### CODE REVIEW

**Comment by kauffj** [comment]
thanks @ioancole, this will get looked at shortly

would be interested in making more contributions here? you can also email me at jeremy@lbry.com

File: app/actions/auto_complete.go [hunk 1]
```diff
@@ -44,17 +44,28 @@ func AutoComplete(r *http.Request) api.Response {
...
-	mmName := elastic.NewMultiMatchQuery(acRequest.S).
-		Type("phrase_prefix").Slop(5).MaxExpansions(50).
-		Field("name^4")
-	query := elastic.NewBoolQuery().Should(nested, mmName)
+		Type("phrase_prefix").Slop(5).MaxExpansions(50)
+
+	query := elastic.NewBoolQuery()
+
+    if acRequest.S[0] == '@' {
+        mmATD = mmATD.Field("value.Claim.stream.metadata.author")
```
Comment (tiger5226) [inline app/actions/auto_complete.go:52]: you want to use the claim name not the author field.

**Review by tiger5226 — CHANGES_REQUESTED** [review]

**Comment by tiger5226** [comment]
https://github.com/lbryio/lighthouse/blob/master/app/actions/search/query.go#L182 here is an area in the search query where we do something similar if you want to take another stab at it.

**Comment by ioancole** [comment]
Yes fair enough, thanks!

**Comment by kauffj** [comment]
@ioancole still planning to finish this?

**Review by tiger5226 — APPROVED** [review]

**Pull request merged**